### PR TITLE
Add tooling to simplify the workflow for command extensions

### LINKED ISSUE #12600: Add basic tooling to support Spack command extensions
As a user I want a command to help managing Spack command extensions so that I don't have to perform repetitive tasks manually

### Rationale

Since a while we have an experimental support in Spack for [custom extensions](https://spack.readthedocs.io/en/latest/extensions.html). What I propose in this issue is to build a bit of tooling to make it easier to use them. The bare minimum I think would be sub-commands to:

- Help creating the structure for new extensions
- Add and remove existing extensions without having to fetch code or modify YAML files manually

### Description

One possibility is to repurpose the `spack extensions` command and rename it `spack package-extensions`:
```console
$ spack package-extensions -h
usage: spack package-extensions [-h] [-l | -p | -d] [-s TYPE] [-v VIEW] ...

list extensions for package

positional arguments:
  spec                  spec of package to list extensions for

optional arguments:
  -h, --help            show this help message and exit
  -l, --long            show dependency hashes as well as versions
  -p, --paths           show paths to extension install directories
  -d, --deps            show full dependency DAG of extensions
  -s TYPE, --show TYPE  one of packages, installed, activated, all
  -v VIEW, --view VIEW  the view to operate on
```
If we do that then we can add subcommands to `spack extensions` to create a new custom extension:
```console
$ spack extensions create <folder>
```
to fetch and use an extension hosted somewhere (e.g. on github):
```console
$ # Fetch and registers an extension hosted at url
$ spack extensions get <url>
```
or to remove it:
```console
$ spack extensions remove <url>
```
Would these additions be welcome if coded?

### PR DESCRIPTION
**Pull request by alalazo**
fixes #12600

This PR adds a new command to simplify working with Spack extensions. Documentation is added to highlight its use and show common scenarios.

Current roadmap for this PR:

- [x] Add commands to manage command extensions from the command line
- [x] Extend the documentation section on "Custom extensions"
- [ ] Add tests for the new feature, consolidate the code

### CODE REVIEW

**Comment by alalazo** [comment]
@tgamblin I wonder what are your thoughts on the cli and docs. Also, going forward am I allowed to repurpose the current `extensions` command and rename it `package-extensions`?

File: lib/spack/spack/cmd/command_extensions.py [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+                    tty.debug(msg.format(folder))
+
+    if not found:
+        msg = 'Extension "{0}" is not installed'
+        tty.msg(msg.format(args.name))
+
+
+def command_extensions_update(args):
+    for scope in spack.config.scopes():
+        # Skip default scopes when updating as those are meant to be red-only
```
Comment (chissg) [inline lib/spack/spack/cmd/command_extensions.py]: red -> read.

File: lib/spack/spack/cmd/command_extensions.py [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+            repository_dir = os.path.join(ext['root'], name)
+            if ext2ver and name not in ext2ver:
+                continue
+
+            updated = True
+            # The call below is needed to deal with the old string format
+            ensure_url_and_version(ext)
+
+            # Check if we have a specific version to check out
+            version = ext2ver.get(ext['name'], None)
```
Comment (chissg) [inline lib/spack/spack/cmd/command_extensions.py]: See discussion above re default version in `ensure_url_and_version()`.

**Comment by alalazo** [comment]
Close as stale, but I could still revisit the idea sometimes :slightly_smiling_face:

**Pull request closed without merging**